cpp/ps0005/ps0005_solv.cpp

```diff
@@ -4,82 +4,75 @@ using namespace std;
+        
+        if(op == '+')
         {
-          result = x+y;
-          cout<<" ( "<< x <<" ) " << z <<" ( "<< y <<" ) " <<" = "<< result <<" \n ";
+          result = num1 + num2;
+          cout << " ( " << num1 << " ) " << op << " ( " << num2 << " ) " << " = " << result << "\n";
         }
-
-
-        else if(z == '-')
+        
+        else if(op == '-')
         {
-          result = x-y;
-          cout<<" ( "<< x <<" ) " << z <<" ( "<< y <<" ) " <<" = "<< result <<" \n ";
+          result = num1 - num2;
+          cout << " ( " << num1 << " ) " << op << " ( " << num2 << " ) " << " = " << result << "\n";
         }
-
-          else if(z == '*')
+        
+        else if(op == '*')
         {
-          result = x*y;
-          cout<<" ( "<< x <<" ) " << z <<" ( "<< y <<" ) " <<" = "<< result <<" \n ";
+          result = num1 * num2;
+          cout << " ( " << num1 << " ) " << op << " ( " << num2 << " ) " << " = " << result << "\n";
         }
-
-
-          else if(z == '/')
+        
+        else if(op == '/')
         {
-             if(z == '/' && x!=0 && y==0)
+            if(num1 != 0 && num2 == 0)
             {
-              cout<<" Math Error : cannot divide by 0 .  \n ";
+              cout << "Math Error : cannot divide by 0.\n ";
             }
-
-             else if(z == '/' && x==0 && y==0)
+            else if(num1 == 0 && num2 == 0)
             {
-              cout<<" Infinite number of solutions : ( 0 / 0 ) is Undefined Quantity .  \n ";
-
+              cout << "Infinite number of solutions : ( 0 / 0 ) is Undefined Quantity.\n ";
             }
-
-              else
+            else
             {
-              result = x/y;
-              cout<<" ( "<< x <<" ) " << z <<" ( "<< y <<" ) " <<" = "<< result <<" \n ";
+              result = num1 / num2;
+              cout << " ( " << num1 << " ) " << op << " ( " << num2 << " ) " << " = " << result << "\n";
             }
-        }
-
-          else if(z == '%')
+        } // 1/1
+        
+        else if(op == '%')
         {
-              if(z == '%' && ( x== (float)x || y== float(y)))
+            if(num1 != (int)num1 || num2 != (int)num2)
             {
-              cout<<" Cannot execute modulo operation for floating point numbers . \n ";
+              cout << "Cannot execute modulo operation for floating point numbers.\n";
             }
-
-              else
+            else
             {
-              result = x%y;
-              cout<<" ( "<< x <<" ) " << z <<" ( "<< y <<" ) " <<" = "<< result <<" \n ";
+              result = (int)num1 % (int)num2;
+              cout << " ( " << num1 << " ) " << op << " ( " << num2 << " ) " << " = " << result << "\n";
             }
-
-        }
-
+        } // 0/1
     }
-
-     else if(z != '+' || z != '-'  ||z != '*' || z != '/' || z != '%')
+    else
     {
       cout<<" Invalid character for operator .  \n ";
-    }
-
-
-
+    } // 1/1
+    // 1/1
+    
     return 0;
 }
+// good work 👍 7.5/10

```

Comment:
- [x] program is in the correct path
- [x] no syntax errors while compilation
- [ ] formatted code to be clear and readable
- [ ] no bugs or warnings while execution
- [ ] commented some phrases for blocks of code
- [ ] main returns non-zero/zero number according to the case
+1 bonus mark for your effort ⭐

you can check the [problem solution](https://onlinegdb.com/2mF7i4R8H) for more